src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11163,6 +11201,15 @@ private ExpressionSyntax ParseTermWithoutPostfix(Precedence precedence)
                             {
                                 return ParseDeclarationExpression(ParseTypeMode.Normal, isScoped: false);
                             }
+                            else if (IsCurrentTokenFieldInKeywordContext() && PeekToken(1).Kind != SyntaxKind.ColonColonToken)
+                            {
```

Comment:
>{ [](http://example.com/codeflow?start=28&length=1)

Should report `ErrorCode.INF_IdentifierConflictWithContextualKeyword` here rather than in the binder, and should be reported regardless of whether the `field` token is treated as an identifier or a keyword.